app/js/user.js

```diff
@@ -88,6 +108,50 @@
           });
         });
       });
+    },
+
+    /**
+     * Updates the local user data as well as last sync time.
+     * @param {object} user_data - The ammended user data and settings i.e.
+     * user_data = {
+     *   user: user,
+     *   settings, settings
+     * };
+     * @returns The updated user object from indexedDB
+     */
+    update_user: function(user_data) {
+      SumoDB.update_user(user_data).then(function(user) {
+        var normalized_user = normalize_user(user);
+        normalized_user.last_sync = Date.now();
+        return asyncStorage.setItem('user', normalized_user).then(function() {
+          return user;
```

Comment:
Would what is returned from the async call not actually return the normailzed user? That is what we set.